azure/scope/machine.go

```diff
@@ -250,15 +261,19 @@ func (m *MachineScope) NICSpecs() []azure.NICSpec {
 		spec.PublicIPName = azure.GenerateNodePublicIPName(m.Name())
 	}
 
-	return []azure.NICSpec{spec}
+	if m.cache != nil {
+		spec.SKU = &m.cache.networkInterfaceSKU
+	}
+
+	return []azure.ResourceSpecGetter{spec}
 }
 
 // NICIDs returns the NIC resource IDs.
 func (m *MachineScope) NICIDs() []string {
 	nicspecs := m.NICSpecs()
 	nicIDs := make([]string, len(nicspecs))
 	for i, nic := range nicspecs {
-		nicIDs[i] = azure.NetworkInterfaceID(m.SubscriptionID(), m.ResourceGroup(), nic.Name)
+		nicIDs[i] = azure.NetworkInterfaceID(m.SubscriptionID(), m.ResourceGroup(), nic.ResourceName())
```

Comment:
```suggestion
		nicIDs[i] = azure.NetworkInterfaceID(m.SubscriptionID(), nic.ResourceGroupName(), nic.ResourceName())
```

should be equivalent but this is slightly more precise, and more resilient if we ever support having the NIC in a separate resource group